### ISSUE TITLE
Drivers for pre-retina MacBooks

### ISSUE THREAD

**Issue by mariobrostech** [arnoldthebat/chromiumos#210]
Hello,
I have a MacBook 4,1 (Early 2008) model that I would like to use Chromium OS on because it is fairly lightweight.

I installed the latest "special" build, but the problem is that the trackpad won't work at all.
This is a dealbreaker for me. Is there any way you can embed the driver into a future release or that I can install it myself?

Thank you in advance for your help!

**Comment by robarago** [comment]
Hi,
Here is another MacBook (this time 2,1) user with the same issue. The trackpad is working in special 58.8872 but not in any newer build.

I'd really appreciate that the new special bundle brings mac trackpad to life again!
Thank you very much for your excellect work.

**Comment by mariobrostech** [comment]
> Hi,
> Here is another MacBook (this time 2,1) user with the same issue. The trackpad is working in special 58.8872 but not in any newer build.
> 
> I'd really appreciate that the new special bundle brings mac trackpad to life again!
> Thank you very much for your excellect work.

I'll try this build over the weekend... I haven't had success with any of the special builds, so I'm excited to get it working (even if it's on an outdated version).

**Comment by mariobrostech** [comment]
> Hi,
> Here is another MacBook (this time 2,1) user with the same issue. The trackpad is working in special 58.8872 but not in any newer build.
> 
> I'd really appreciate that the new special bundle brings mac trackpad to life again!
> Thank you very much for your excellect work.

Hi, is there anything in particular that you did to get the touchpad working on R58? I can't get it working on my machine... Any thoughts?

**Comment by robarago** [comment]
Hi,

I didn't do anything special, but I've seen the trackpad only works in Cx86OS_R58-8872.B-Special.7z , not in Cx86OS_R58-9334.B-Vanilla.7z, check it out. I've got also Flash player working in this version thanks to [this script](https://gist.github.com/kirbyfan64/07c19d007c50f21291c5d1dec041fdc6), if you're interested.

Also check that you'll need two click once or twice the ENTER key after resuming the laptop to recover the pointer.

Regards.

**Comment by mariobrostech** [comment]
> Hi,
> 
> I didn't do anything special, but I've seen the trackpad only works in Cx86OS_R58-8872.B-Special.7z , not in Cx86OS_R58-9334.B-Vanilla.7z, check it out. I've got also Flash player working in this version thanks to [this script](https://gist.github.com/kirbyfan64/07c19d007c50f21291c5d1dec041fdc6), if you're interested.
> 
> Also check that you'll need two click once or twice the ENTER key after resuming the laptop to recover the pointer.
> 
> Regards.

@robarago Hmmm... I tried the Camd64OS_R58-8872.B-Special.7z file - I doubt the x86 version would make much of a difference, but I'll give it a shot. can you send a link to your exact MacBook on [EveryMac](https://everymac.com/)? I am planning on using chromefy to 'upgrade' this to the official Chrome OS, so I don't care if it's an outdated version of Chromium - I just need the drivers working before I do the upgrade. Any other ideas?

**Comment by robarago** [comment]
Hi @mariobrostech,

I'm afraid x86 makes a big difference (in my case, at least). I've been unable to run any x86_64 image of ChromeOS because my machine ([MacBook2,1](https://everymac.com/systems/apple/macbook/specs/macbook-core-2-duo-2.16-white-13-mid-2007-specs.html)) had EFI32 firmware in a 64 bit processor (yes, very sad!).

Check also [this link](https://bbs.archlinux.org/viewtopic.php?id=121153) given by arnoldthebat (distribution maintainer) in [this forum](https://github.com/arnoldthebat/chromiumos/issues/10) to a Mac user regarding touchpad tap not working, it might be useful too.

Also I can email you my `dmesg` output and the `20-touchscreen.conf` configuration file for reference if you wish. I've made no modifications to this later, it's the distribution original config file, but it doesn't seem to give too much info.

Good luck!

**Comment by mariobrostech** [comment]
@robarago Okay, I'll be sure to try this.
Thanks again for all your help!

**Comment by mariobrostech** [comment]
@robarago 
**I cannot get this working on more modern versions.**
Below is what I tried (I even wrote it into a guide as I tried it). Nothing has worked for me so far.

_`sudo su`
`mount -o remount, rw /`
`cd /etc/gesture/`
`vi 40-touchpad-cmt.conf`
_vi is very complex. [This tutorial](https://www.washington.edu/computing/unix/vi.html) should help you in navigating it and modifying this file._

Modify the one with Identifier "touchpad" to match these values:
`Section "InputClass"`
 `Identifier "touchpad"`
 `MatchIsTouchpad "on"`
 `MatchDevicePath "/dev/input/event*"`
 `Driver "synaptics"`
 `Option "AccelerationProfile" "-1"`
 `Option "Scroll Buttons" "0"`
 `Option "Scroll Axes" "1"`
 `Option "Scroll X Out Scale" "1.25"`
 `Option "Scroll Y Out Scale" "1.25"`
`EndSection`

[If that didn't work, try this slightly longer command listed on this page.](https://chromium.arnoldthebat.co.uk/files/fw/etc/X11/xorg.conf.d/50-touchpad-cmt.conf)

If you want to do further tweaking, you'll need to modify these values further. As a good starting point, modify the values in this newly created file to match the ones listed on [this page](https://wiki.archlinux.org/index.php/Touchpad_Synaptics#Frequently_used_options)._
